### PR TITLE
split out some more apis to allow for more extensive hook testing

### PR DESCRIPTION
**Pull request by novas0x2a**
The goal of this, overall, is to make it possible to write tests that more extensively involve hooks; this involves a bit more unification of the tiller/non-tiller code paths (to allow rendering to be more similar between the two) and a few new exports.

### CODE REVIEW

**Comment by novas0x2a** [comment]
Hm, what's the status of this? Any comments, positive or negative? :D

**Comment by helgi** [comment]
@bacongobbler any chance of this potentially going into 2.14?

File: pkg/manifest/hooks.go [hunk 1]
```diff
@@ -27,34 +28,21 @@ import (
...
-	hooks.BeforeHookCreation: release.Hook_BEFORE_HOOK_CREATION,
-}
+// NOTESFILE_SUFFIX that we want to treat special. It goes through the templating engine
+// but it's not a yaml file (resource) hence can't have hooks, etc. And the user actually
+// wants to see this file after rendering in the status command. However, it must be a suffix
+// since there can be filepath in front of it.
+const notesFileSuffix = "NOTES.txt"
 
-// Manifest represents a manifest file, which has a name and some content.
-type Manifest = manifest.Manifest
```
Comment (bacongobbler) [inline pkg/manifest/hooks.go]: Why was this removed?

**Comment by bacongobbler** [comment]
The fake client has been completely removed in Helm 3, so I'm not too sure how useful this PR would be going forward for Helm 2. I could see some value porting some of these changes over to Helm 3, though. Once #5365 is reconciled and merged, would you be interested in porting these changes over to Helm 3?

@helgi what value do you see from this PR? Are you just interested in the refactored code exposing more Helm internals?

**Comment by helgi** [comment]
@bacongobbler yeah it helps a lot (and we use this for the same internal project :-)

PR might not be the right venue for this but are there any documented timelines for v3 to see if it makes sense to do a v3 only variant?

@novas0x2a has a better idea on the possibility of the forward port and such

**Comment by bacongobbler** [comment]
No documented timeline for a v3 release at this point.

**Comment by novas0x2a** [comment]
@bacongobbler the core purpose of this was to split out enough stuff to write a test for the release testing; i have a RunReleaseTest implementation that uses a kube fake (`k8s.io/client-go/kubernetes/typed/core/v1/fake.Clientset`) to launch pseudo-real pods when the helm tests are triggered in the mock ecosystem. (I haven't submitted that here because it's a pretty big departure from what the current test mock is doing, but i'd be glad to stick it somewhere in the repo if you want it)

File: pkg/manifest/hooks.go (hunk 1 above)
Comment (novas0x2a) [reply pkg/manifest/hooks.go]: It shouldn't have been, sorry, I rebased and made sure I left the breadcrumbs behind for things that were exported that I moved out of the tiller module.

**Comment by novas0x2a** [comment]
A few other things fell out of the above purpose, so I fixed them as I noticed them; this set of changes also makes `helm template` and the RenderMock code use the same render path as the tiller does, so there should be less surprises there

**Comment by novas0x2a** [comment]
Rebased upon current master (the branch name is unchanged to avoid closing the PR)

**Comment by novas0x2a** [comment]
Hm, the circleci build failed, but it kind of looks like it was timed out? Retriggering with a force push...

**Comment by novas0x2a** [comment]
Is it possible to get an ACK/NAK/Comments on this? Keeping it up to date has been a fair amount of effort, and I don't really want to do it anymore if there's no chance of this landing.

**Comment by bacongobbler** [comment]
Hey @novas0x2a, it doesn't look like there's any interest from the maintainers or the community on this PR, especially since we've been focusing on Helm 3 recently. Sorry there hasn't been much of a response from us; we've been swamped with getting Helm 3 out the door.
